src/composables/fungibleTokens.ts

```diff
@@ -0,0 +1,491 @@
+  categorizeContractCallTxObject,
+  convertToken,
+  CURRENCY_URL,
+  fetchJson,
+  handleUnknownError,
+  watchUntilTruthy,
+  ZEIT_TOKEN_INTERFACE,
+} from '../popup/utils';
+import type {
+  IDefaultComposableOptions,
+  IDexContracts,
+  IDexTokenPairs,
+  INetwork,
+  IPendingTransaction,
+  IRawTokenTransaction,
+  IToken,
+  ITokenList,
+  ITransaction,
+  IAsset,
+  Balance,
+} from '../types';
+import { createPollingBasedOnMountedComponents } from './composablesHelpers';
+import { useSdk } from './sdk';
+
+const activeAddress = ref<string | null>(null);
+const availableTokens = ref<ITokenList>({});
+const tokens = ref<Record<string, {tokenBalances: IToken[]}>>({});
+const aePublicData = ref<IAsset | null>(null);
+const transactions = ref<Record<string, {transactions: ITransaction[]}>>({});
+
+const initPollingWatcher = createPollingBasedOnMountedComponents();
+
+export interface ITransferTokensOptions {
+  contractId: string
+  recipient: string
+  amount: number
+  options: {
+    waitMined: boolean,
+    modal: boolean
+  }
+}
+
+export interface IBurnTriggerPosOptions {
+  contractId: string
+  amount: number
+  posAddress: string
+  invoiceId: string
+  options: {
+    waitMined: boolean,
+    modal: boolean
+  }
+}
+
+export function useFungibleTokens({ store }: IDefaultComposableOptions) {
+  const activeNetwork = computed<INetwork>(() => store.getters.activeNetwork);
+  const getDexContracts = computed<IDexContracts>(() => store.getters.getDexContracts);
+  const middleware = computed(() => store.state.middleware);
+  const activeCurrency = computed(() => store.state.current.currency);
+  const { getSdk } = useSdk({ store });
+
+  function getTokenBalances(address: string) {
+    return tokens.value?.[address]?.tokenBalances || [];
+  }
+
+  const tokenBalances = computed((): IToken[] => (
+    activeAddress.value
+      ? getTokenBalances(activeAddress.value)
+      : []
+  ));
+
+  function resetTokensAndTransactions() {
+    tokens.value = {};
+    transactions.value = {};
+  }
+
+  /**
+   * Returns an object representing the Aeternity token,
+   * with the specified balance and balance currency.
+   * If `aePublicData` is not empty, its values are included in the returned object.
+   *
+   * @param {Object} options - An object with the following properties:
+   * @param {string} options.balanceCurrency - The balance currency of the token.
+   * @param {number} options.tokenBalance - The balance of the token.
+   * @returns {Object} - An object representing the Aeternity token, with the following properties:
+   *  - convertedBalance: The balance of the token.
+   *  - symbol: The symbol of the token (i.e. "AE").
+   *  - balanceCurrency: The balance currency of the token.
+   *  - contractId: The contract ID of the Aeternity token contract.
+   */
+  function getAeternityToken({
+    balanceCurrency,
+    tokenBalance,
+  }: { balanceCurrency: BigNumber | number, tokenBalance: BigNumber | Balance }): IToken | IAsset {
```

Comment:
This function returns one structure so the return type `IToken | IAsset` is not true.